pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -260,10 +260,16 @@ func (r *reconciler) ensureLoadBalancerService(ci *operatorv1.IngressController,
 		return false, nil, err
 	}
 
+	// BZ2054200: Don't modify/delete services that are not directly owned by this controller
+	ownLBS := isServiceOwnedByIngressController(currentLBService, ci)
+
 	switch {
 	case !wantLBS && !haveLBS:
 		return false, nil, nil
 	case !wantLBS && haveLBS:
+		if !ownLBS {
+			return false, nil, fmt.Errorf("refusing to delete load balancer service not owned by ingress controller: %s", controller.LoadBalancerServiceName(ci))
```

Comment:
I think this wording is a little clearer:
```suggestion
			return false, nil, fmt.Errorf("a conflicting load balancer service exists that is not owned by the ingresscontroller: %s", controller.LoadBalancerServiceName(ci))
```
Likewise for the other 3 similar error messages.

Reply:
Agreed. Done.